src/industry_gui.cpp

```diff
@@ -840,33 +867,35 @@ class IndustryViewWindow : public Window
 			ir.top += FONT_HEIGHT_NORMAL + WidgetDimensions::scaled.vsep_wide;
 		}
 
-		CargoSuffix cargo_suffix[lengthof(i->accepts_cargo)];
-		GetAllCargoSuffixes(CARGOSUFFIX_IN, CST_VIEW, i, i->type, ind, i->accepts_cargo, cargo_suffix);
 		bool stockpiling = HasBit(ind->callback_mask, CBM_IND_PRODUCTION_CARGO_ARRIVAL) || HasBit(ind->callback_mask, CBM_IND_PRODUCTION_256_TICKS);
 
-		for (byte j = 0; j < lengthof(i->accepts_cargo); j++) {
-			if (!IsValidCargoID(i->accepts_cargo[j])) continue;
+		for (const auto &a : i->accepts) {
+			if (!IsValidCargoID(a.cargo)) continue;
 			has_accept = true;
 			if (first) {
 				DrawString(ir, STR_INDUSTRY_VIEW_REQUIRES);
 				ir.top += FONT_HEIGHT_NORMAL;
 				first = false;
 			}
-			SetDParam(0, CargoSpec::Get(i->accepts_cargo[j])->name);
-			SetDParam(1, i->accepts_cargo[j]);
-			SetDParam(2, i->incoming_cargo_waiting[j]);
+
+			CargoSuffix suffix;
+			GetCargoSuffix(CARGOSUFFIX_IN, CST_VIEW, i, i->type, ind, a.cargo, &a - i->accepts.data(), suffix);
```

Comment:
`GetCargoSuffix` returning `CargoSuffix` would be nicer, but I guess that's also a lot of extra work for this PR making it even harder to review. So more an idea for a future improvement (or at least a separate commit).